### ISSUE TITLE
Specify the name of the training stream in the config

### ISSUE THREAD

**Issue by petrbel** [Cognexa/cxflow#190]
It would be pretty nice to have a single dataset with let's say two trainign streams (generative and regular one). One would be able to pre-train the model on the generated data and then (in the separate training) finetune it on the real data.

Implementing this would make cxflow be more consistent with other streams that are easily configurable.

Config suggestion:
```yaml
main_loop:
  train_stream: train  # or generative_train
```

What do you think?

**Comment by FloopCZ** [comment]
Looks nice, I agree.

**Comment by blazekadam** [comment]
What about multiple train streams support?

**Comment by FloopCZ** [comment]
@blazekadam That sounds good as well! How would you do it? Would you concatenate/interleave the streams? Or would it be possible to specify how long should we take the data from the first stream and how long from the second, ... ?

**Comment by petrbel** [comment]
Many possibilities... I think the best solution is to implement e.g. `BaseInterleavingDataset` etc which would wrap the actual datasets and handls the policy of their train streams usage.

**Comment by FloopCZ** [comment]
Yes, but in such a case, we would not see the statistics for the multiple streams separately (loss, accuracy, ...). For that, we would probably need to support it directly in the main loop. But it is a public API change (change of the config file format), so I believe we should think what the final situation looks like.
